test/runtime/fqdn.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package RuntimeTest
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/cilium/cilium/pkg/fqdn"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+
+	. "github.com/onsi/gomega"
+)
+
+var bindTemplate = `
+$TTL 3
+$ORIGIN cilium.test.
+
+@       IN      SOA     cilium.test. admin.cilium.test. (
+                        200608081       ; serial, todays date + todays serial #
+                        8H              ; refresh, seconds
+                        2H              ; retry, seconds
+                        4W              ; expire, seconds
+                        1D )            ; minimum, seconds
+;
+;
+@               IN NS server
+server.cilium.test. IN A 127.0.0.1
+
+world1.cilium.test. IN A %[1]s
+world2.cilium.test. IN A %[2]s
+world3.cilium.test. IN A %[3]s
+
+roundrobin.cilium.test.    1   IN   A %[1]s
+roundrobin.cilium.test.    1   IN   A %[2]s
+roundrobin.cilium.test.    1   IN   A %[3]s
+`
+
+var bind9ZoneConfig = `
+zone "cilium.test" {
+        type master;
+        file "/etc/bind/db.cilium.test";
+};
+`
+
+var _ = Describe("RuntimeFQDNPolicies", func() {
+	const (
+		bindContainerImage = "eloycoto/docker-bind"
```

Comment:
add this to the cilium dockerhub, specify a version and write the repository where this is installed. `docker.io/...`